src/models/TranslationKey/deleteTranslationKey/index.js

```diff
@@ -0,0 +1,27 @@
+// @flow
+import TranslationKey from '@models/TranslationKey'
+
+export type DeleteTranslationKeyArgs = {
+  translationKeyId: string,
+  ownerId: string
+}
+
+const deleteTranslationKey = async ({
+  translationKeyId,
+  ownerId
+}: DeleteTranslationKeyArgs): Promise<void> => {
+  const translationKey: TranslationKey = await TranslationKey.query()
+    .eager('project')
+    .findById(translationKeyId)
+
+  if (
+    !translationKey ||
+    (translationKey.project && !translationKey.project.hasOwnerId(ownerId))
+  ) {
+    throw new Error('This translation key was not found.')
+  }
+
+  await TranslationKey.query().deleteById(translationKeyId)
```

Comment:
Maybe guard this with a try/catch statement so you can throw a generic error in case DB query fails ?

Reply:
Well it should only fail if the translation key doesn’t exist but we already throw for that, I’m not sure we should handle the error (ideally we should yes) :~}